conn/conn.go

```diff
@@ -58,23 +60,24 @@ func New(param ConnectionParam, maxIdle int) (*redis.Pool, ReconnectPolicy) {
 // considered useless.
 func connect(param ConnectionParam) func() (redis.Conn, error) {
 	return func() (cnx redis.Conn, err error) {
-		if param.Timeout > 0 {
-			cnx, err = redis.DialTimeout("tcp", param.Address,
-				param.Timeout, param.Timeout, param.Timeout)
-		} else {
-			cnx, err = redis.Dial("tcp", param.Address)
+		options := make([]redis.DialOption, 0)
+		if param.UseTLS {
+			options = append(options, redis.DialUseTLS(param.UseTLS))
 		}
 
-		if err != nil {
-			return
+		if param.Password != "" {
+			options = append(options, redis.DialPassword(param.Password))
 		}
 
-		if param.Password != "" {
-			if _, err = cnx.Do("AUTH", param.Password); err != nil {
-				return
-			}
+		if param.Timeout > 0 {
+			options = append(options, []redis.DialOption{
+				redis.DialConnectTimeout(param.Timeout),
+				redis.DialReadTimeout(param.Timeout),
+				redis.DialWriteTimeout(param.Timeout),
+			}...)
 		}
 
+		cnx, err = redis.Dial("tcp", param.Address, options...)
 		return
```

Comment:
I'm not sure that this is needed but it was in the original code. Perhaps something with the way Go handles named return values? If someone can verify it's not needed I can remove it.